sklearn/cluster/hierarchical.py

```diff
@@ -696,7 +717,15 @@ def fit(self, X, y=None):
                              " instance, got 'memory={!r}' instead.".format(
                                  type(memory)))
 
-        if self.n_clusters <= 0:
+        if (self.distance_threshold and self.n_clusters) or \
+                (not self.n_clusters and not self.distance_threshold):
+            raise ValueError("Either n_clusters (>0) or distance_threshold "
+                             "needs to be set, got n_clusters={} and "
+                             "distance_threshold={} instead. "
+                             "Please set n_clusters=None to continue.".format(
```

Comment:
That was how I first attempted it, but I get a few errors from estimator_checks

`
AssertionError: Estimator FeatureAgglomeration should not change or mutate  the parameter n_clusters from None to 2 during fit.
`

Do I have to try and get around this error? Or force the user to set `n_clusters` as None when using `distance_threshold`?